ttnn/cpp/ttnn/compiler_interface/compiler_interface.hpp

```diff
@@ -0,0 +1,31 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <tuple>
+
+#include "ttnn/compiler_interface/singleton_device_context.hpp"
+#include "ttnn/cpp/ttnn/tensor/enum_types.hpp"
+#include "ttnn/tensor/types.hpp"
+
+namespace ttnn::compiler_interface {
+
+using OperandParams =
+    std::tuple<ttnn::SimpleShape, tt::tt_metal::DataType, tt::tt_metal::Layout, tt::tt_metal::MemoryConfig>;
+
+std::tuple<bool, size_t, size_t, size_t> extract_data_from_trace(const nlohmann::json& trace) {
```

Comment:
I suggest to make is a struct. And because what you extract from trace does not extract bool (run success) from the trace, please remove this value from this struct. If you need a struct with run status  - lets have two separate structs like (i'll go with terrible name for clarity, need to find proper names).

```
struct OperationRunMemoryImpact {
  size_t cb_peak_size_per_core = 0;
  size_t l1_buffers_peak_per_core = 0;
  size_t l1_output_buffer_per_core = 0;
};

struct OperationRunQueryResponse {
  OpRunStatus status = OpRunStatus::Error; // (Success, Error (assume InvalidConfiguration))
  OperationRunMemoryImpact memory_impact;
};
```

Reply:
Can you share on why do you need to distinguish between CB and non-cb?

Reply:
they are allocated from different part of the heap. to enable early work on tracking fragmentation.

Reply:
re naming, I went with 

enum class ExecutionStatus { Success, Error };

struct ResourceUsage {
    size_t cb_peak_size_per_core = 0;
    size_t l1_buffers_peak_per_core = 0;
    size_t l1_output_buffer_per_core = 0;
};

struct QueryResponse {
    ExecutionStatus status = ExecutionStatus::Error;
    ResourceUsage resource_usage;
    std::optional<std::string> error_message;
}; 

please let me know if you get better inspiration